packages/lib/src/instrumentation.ts

```diff
@@ -80,19 +90,74 @@ export function init(options: initOptions) {
   }
 }
 
+export function eagerlyPushMetricsIfConfigured() {
+  if (!IS_EAGERLY_PUSHED) {
+    return;
+  }
+
+  if (exporter instanceof PeriodicExportingMetricReader) {
+    logger("Pushing metrics to gateway");
+    _pushMetrics();
+  }
+}
+
+// TODO - add a way to stop the push interval
+// TODO - improve error logging
+// TODO - allow custom fetch function to be passed in
+// TODO - allow configuration of timeout for fetch
 async function pushToGateway(gateway: string) {
-  const exporterResponse = await exporter.collect();
-  const serialized = new PrometheusSerializer().serialize(
-    exporterResponse.resourceMetrics,
-  );
-
-  await fetch(gateway, {
-    method: "POST",
-    mode: "cors",
-    body: serialized,
-  });
-  // we flush the metrics at the end of the submission to ensure the data is not repeated
-  await exporter.forceFlush();
+  try {
+    const exporterResponse = await exporter.collect();
+
+    const serialized = new PrometheusSerializer().serialize(
+      exporterResponse.resourceMetrics,
+    );
+
+    if (typeof fetch === "undefined") {
+      console.error(
+        "Fetch is undefined, cannot push metrics to gateway. Consider adding a global polyfill.",
+      );
+      return;
+    }
+
+    try {
+      const response = await fetch(gateway, {
+        method: "POST",
+        mode: "cors",
+        body: serialized,
+      });
+      if (!response.ok) {
+        console.error(
+          `Error pushing metrics to gateway: ${response.statusText}`,
+        );
+        // NOTE - Uncomment to log the response body
+        // console.error(JSON.stringify(await response.text(), null, 2));
+      }
+    } catch (fetchError) {
+      console.error(
+        `Error pushing metrics to gateway: ${
+          fetchError?.message ?? "<no error message found>"
+        }`,
+      );
+    }
+
+    try {
+      // we flush the metrics at the end of the submission to ensure the data is not repeated
+      await exporter.forceFlush();
+    } catch (error) {
+      console.error(
+        `Error flushing metrics after push: ${
+          error?.message ?? "<no error message found>"
+        }`,
+      );
+    }
+  } catch (collectError) {
```

Comment:
Why the nested try-catches? If you expect this catch to only apply to the collecting part, why not place it *before* the other two try-catch blocks?